nep-5.mediawiki

```diff
@@ -91,6 +99,8 @@ The function SHOULD check whether the <code>from</code> address equals the calle
 
 If the <code>to</code> address is a deployed contract, the function SHOULD check the <code>payable</code> flag of this contract to decide whether it should transfer the tokens to this contract.
 
+The function MUST call <code>onPayment</code> method AFTER fire the <code>Transfer</code> event. This call need the flag <code>IfExists</code> and the arguments <code>from</code>, <code>to</code>, <code>amount</code>.
```

Comment:
This conflicts with the existing description of the `transfer` method.

> If the method succeeds, it MUST fire the transfer event, and MUST return true, even if the amount is 0, or from and to are the same address.

We can't say if `transfer()` succeeds it must fire the `transfer` event, but then also say that after the `transfer` event we call `onPayable`, which in its turn is supposed to be able to modify execution according to this text from #108 
> must return True if everything is correct or throw an exception otherwise. 

I think it is supposed to say "must call onPayment BEFORE firing the Transfer event"